source/worker.go

```diff
@@ -0,0 +1,126 @@
+// Copyright © 2024 Meroxa, Inc. and Miquido
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package source
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"time"
+
+	"github.com/conduitio-labs/conduit-connector-elasticsearch/internal/elasticsearch/api"
+	"github.com/conduitio/conduit-commons/opencdc"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+)
+
+type Worker struct {
+	source           *Source
```

Comment:
IMHO, it would be better if `Worker` doesn't depend on `Source`, because it doesn't call any of its methods. If certain values are needed, they can be given to `Worker`. For example, a `Worker` needs an ES client, so we can just give it a client, and not the `Source`, through which it access the client. That way, we keep the internal of `Source` away from worker.

Reply:
Done.